chia/consensus/blockchain.py

```diff
@@ -905,6 +912,7 @@ async def get_block_generator(
             # (as long as we're in the main chain)
             result = await self.block_store.get_generators_at(block.transactions_generator_ref_list)
         else:
+            assert fork is not None
```

Comment:
the if-statement on line 905-908 could probably be simplified to:
```
if fork is None:
```

and maybe the current condition could be an `assert`

Reply:
i almost changed it but realised that would change the logic a bit to decide if this is a reorg by the caller and not in get_block_generator